pkg/kv/kvserver/replica_sst_snapshot_storage.go

```diff
@@ -42,14 +47,26 @@ func NewSSTSnapshotStorage(engine storage.Engine, limiter *rate.Limiter) SSTSnap
 	}
 }
 
+// Init initializes the SSTSnapshotStorage struct.
+func (s *SSTSnapshotStorage) Init() {
```

Comment:
nit: any reason we're not just doing this in `NewSSTSnapshotStorage()`?

Reply:
Good question. The goal is to avoid a Mutex copy as much as possible. The syntax to do it in one line without copying the return value in `NewSSTSnapshotStorage` is somewhat ugly but not as bad as it was in one of my earlier iterations (which had more stuff in `mu`), so I just went ahead and made that change.

Reply:
Thanks! Why are we worried about the mutex copy? We're only calling this once on server startup, and there is no concurrency until it's constructed, but maybe I'm missing something.